src/internal/connector/exchange/attachment.go

```diff
@@ -93,10 +93,12 @@ func uploadLargeAttachment(
 	uploader attachmentUploadable,
 	attachment models.Attachmentable,
 ) error {
-	var (
-		bs   = attachmentBytes(attachment)
-		size = int64(len(bs))
-	)
+	bs, err := GetAttachmentBytes(attachment)
+	if err != nil {
+		return err
```

Comment:
errs returned in a func that is the last receiver of a ctx always need to add the context to the error.
```suggestion
		return clues.Stack(err).WithClues(ctx)
```